ecommerce/api.py

```diff
@@ -634,9 +637,15 @@ def create_unfulfilled_order(user):
                 quantity=basket_item.quantity,
             )
 
+        coupon_version = None
         for coupon_selection in basket.couponselection_set.all():
             coupon = coupon_selection.coupon
-            redeem_coupon(coupon_version=latest_coupon_version(coupon), order=order)
+            coupon_version = latest_coupon_version(coupon)
+            redeem_coupon(coupon_version=coupon_version, order=order)
+        order.total_price_paid = get_product_version_price_with_discount(
+            coupon_version=coupon_version, product_version=product_version
+        )
+        order.save()
```

Comment:
Any specific reason we have successive `save`  and `save_and_log` calls?

Reply:
The `save_and_log` serializes before and after, so this `save` makes it so the audit table starts with a log having the same before and after serialized field. Otherwise the log will show a $0 total price as `data_before` and the actual total price as `data_after`. It's not an important thing either way but wanted to keep it consistent.